src/synapse.ts

```diff
@@ -135,8 +137,34 @@ export class Synapse {
     if (options.pieceRetriever != null) {
       pieceRetriever = options.pieceRetriever
     } else {
-      const chainRetriever = new ChainRetriever(pandoraService)
-      pieceRetriever = new FilCdnRetriever(chainRetriever, network)
+      const chainRetriever = new ChainRetriever(pandoraService /*, no child here */)
+      let underlyingRetriever: PieceRetriever = chainRetriever
+
+      // Handle subgraph piece retriever - can provide either a service or configuration
+      if (options.subgraphService != null || options.subgraphConfig != null) {
+        try {
+          let subgraphService: SubgraphRetrievalService
+
+          if (options.subgraphService != null) {
+            subgraphService = options.subgraphService
+          } else if (options.subgraphConfig != null) {
+            subgraphService = new SubgraphService(options.subgraphConfig)
+          } else {
+            // This shouldn't happen due to the if condition above, but TypeScript doesn't know that
+            throw new Error('Invalid subgraph configuration: neither service nor config provided')
+          }
+
+          underlyingRetriever = new SubgraphRetriever(subgraphService, chainRetriever)
```

Comment:
How about this, does TS complain at this?

```suggestion
          const subgraphService: SubgraphRetrievalService =
            options.subgraphService ??
            (options.subgraphConfig ? new SubgraphService(options.subgraphConfig) : undefined)!;
      
          underlyingRetriever = new SubgraphRetriever(subgraphService, chainRetriever);
```

Reply:
Yes, ts-eslint will complain for the use of `non-null-assertion` - `(!)`